### PR TITLE
Applying integration of NastyMPI CI build

### PR DESCRIPTION
**Pull request by fuchsto**

### CODE REVIEW

File: dash/include/dash/Array.h [hunk 1]
```diff
@@ -1014,7 +1066,10 @@ class Array
   void barrier() const
   {
     DASH_LOG_TRACE_VAR("Array.barrier()", m_team);
-    if (nullptr != m_team && dash::Team::Null() != *m_team) {
+    if (nullptr != m_globmem) {
+      m_globmem->flush_all();
+    }
+    if (nullptr != m_team && *m_team != dash::Team::Null()) {
```
Comment (devreal) [inline dash/include/dash/Array.h:1072]: Can m_team ever be a nullptr? It seems to be initialized from a dash::Team reference in all c'tors.

Comment (fuchsto) [reply dash/include/dash/Array.h:1072]: Currently yes, but it's a pointer, so I check it. Members can be specified by *lots* of operations and checking for `nullptr` is more robust than validating every possible member assignment.
Also, it's in `Array::barrier()` so it's not like the test will have a say in performance.

Comment (fuchsto) [reply dash/include/dash/Array.h:1072]: ... also, most lint tools would otherwise nag about this.

**Comment by fuchsto** [comment]
@devreal @fmoessbauer @rkowalewski 

See my comments in #184 

Please review and merge.
Please do not use squash merge as every commit refers to a separate logical block of changes.

**Comment by devreal** [comment]
No objections from my side.

File: dash/test/CopyTest.cc [hunk 2]
```diff
@@ -358,19 +358,40 @@ TEST_F(CopyTest, BlockingLocalToGlobalBlock)
...
   // Assign initial values: [ 1000, 1001, 1002, ... 2000, 2001, ... ]
   for (auto l = 0; l < num_elem_per_unit; ++l) {
-    array.local[l] = 0;
+    array.local[l] = ((dash::myid() + 1) * 10000) + (l * 10);
     local_range[l] = ((dash::myid() + 1) * 1000) + l;
   }
   array.barrier();
 
+  // Block- and global offset of target range:
+  auto block_offset  = _dash_size - 1 - dash::myid();
```
Comment (fmoessbauer) [inline dash/test/CopyTest.cc:371]: This holds until #186 is merged

**Review by fmoessbauer — CHANGES_REQUESTED** [review]
Looks Ok to me. However calculations involving `dash::myid()` have to be explicit casted to e.g. `int`. And `GlobMemTest.ConstructorInitializerList` fails in CI

File: dash/test/CopyTest.cc (hunk 2 above)
Comment (fuchsto) [reply dash/test/CopyTest.cc:371]: Requested change is in scope of #186, not this PR

**Comment by rkowalewski** [comment]
Looks fine to me!

**Pull request merged**